core/coreiface/unixfs.go

```diff
@@ -81,7 +80,48 @@ type UnixfsAPI interface {
 	// to operations performed on the returned file
 	Get(context.Context, path.Path) (files.Node, error)
 
-	// Ls returns the list of links in a directory. Links aren't guaranteed to be
-	// returned in order
-	Ls(context.Context, path.Path, ...options.UnixfsLsOption) (<-chan DirEntry, error)
+	// Ls returns the list of links in a directory. Links aren't guaranteed to
+	// be returned in order. If an error occurs, the DirEntry channel is closed
+	// and an error is output on the error channel. Both channels are closed if
+	// the context is canceled.
+	//
+	// Example:
+	//
+	//	dirs, errs := Ls(ctx, p)
+	//	for dirEnt := range dirs {
+	//		fmt.Println("Dir name:", dirEnt.Name)
+	//	}
+	//	err := <-errs
+	//	if err != nil {
+	//		return fmt.Errorf("error listing directory: %w", err)
+	//	}
+	Ls(context.Context, path.Path, ...options.UnixfsLsOption) (<-chan DirEntry, <-chan error)
+}
+
+// LsIter returns a go iterator that allows ranging over DirEntry results.
+// Iteration stops if the context is canceled or if the iterator yields an
+// error.
+//
+// Exmaple:
+//
+//	for dirEnt, err := LsIter(ctx, ufsAPI, p) {
+//		if err != nil {
+//			return fmt.Errorf("error listing directory: %w", err)
+//		}
+//		fmt.Println("Dir name:", dirEnt.Name)
+//	}
+func LsIter(ctx context.Context, api UnixfsAPI, p path.Path, opts ...options.UnixfsLsOption) iter.Seq2[DirEntry, error] {
```

Comment:
This is implemented to operate on any `UnixfsAPI` since their `Ls` should all work the same.  Is this a good assumption / implementation, or should each `UnixfsAPI` provide its own `LsIter`?

Reply:
I guess it doesn't hurt, but it not used?